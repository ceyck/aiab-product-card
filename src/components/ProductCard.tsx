import React, { createContext } from 'react';
import styles from '../styles/styles.module.css';

import { useProduct } from '../hooks/useProduct';
import {
  productContextProps,
  Product,
  OnChangeArgs,
  InitialValues,
  ProductCardHandlers,
} from '../interfaces/interfaces';

export const productContext = createContext({} as productContextProps);

const { Provider } = productContext;

export interface Props {
  // children?: ReactElement | ReactElement[];
  children?: (args: ProductCardHandlers) => JSX.Element;
  product: Product;
  className?: string;
  style?: React.CSSProperties;
  onChange?: (args: OnChangeArgs) => void;
  value?: number;
  initialValues?: InitialValues;
}

export const ProductCard = ({
  children,
  product,
  className,
  style,
  onChange,
  value,
  initialValues,
}: Props) => {
  const {
    counter,
    increaseBy,
    maxCount,
    isMaxCountReached,
    reset,
  } = useProduct({
    onChange,
    product,
    value,
    initialValues,
  });

  return (
    <Provider value={{ counter, increaseBy, product, maxCount }}>
      <div className={`${styles.productCard} ${className}`} style={style}>
        {children
          ? children({
              count: counter,
              isMaxCountReached,
              maxCount: initialValues?.maxCount,
              product,
              increaseBy,
              reset,
            })
          : ''}
      </div>
    </Provider>
  );
};
