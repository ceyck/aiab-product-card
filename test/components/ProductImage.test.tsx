import React from 'react';
import render from 'react-test-renderer';
import { ProductCard, ProductImage } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductImagen', () => {
  test('debe de mostrar la imagen', () => {
    const wrapper = render.create(
      <ProductImage img="https:hola.com" className="custom" />
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  test('no debe de mostrar la imagen', () => {
    const wrapper = render.create(
      <ProductCard product={product2}>
        {() => <ProductImage className="product-image" />}
      </ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
