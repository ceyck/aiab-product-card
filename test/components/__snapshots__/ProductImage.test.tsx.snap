// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImagen debe de mostrar la imagen 1`] = `
<img
  alt="Product"
  className="productImg custom"
  src="https:hola.com"
/>
`;

exports[`ProductImagen no debe de mostrar la imagen 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg product-image"
    src="./coffee-mug.png"
  />
</div>
`;
